ConsoleApp/Program.cs

```diff
@@ -0,0 +1,551 @@
+            arr[^1] = meeting;
+        }
+    }
+    
+    return arr;
+}
+
+static int CountByNameAndRoom((string, DateTime, int, string)[] meetings, string name, string room)
+{
+    int count = 0;
+    
+    foreach ((string name, DateTime start, int duration, string room) meeting in meetings)
+    {
+        if (meeting.name.Equals(name) && meeting.room.Equals(room))
+        {
+            count++;
+        }
+    }
+
+    return count;
+}
+
+static bool IsMeetingNameUnique((string, DateTime, int, string)[] meetings)
+{
+    foreach ((string name, DateTime start, int duration, string room) meeting in meetings)
+    {
+        int count = CountByNameAndRoom(meetings, meeting.name, meeting.room);
+        
+        if (count > 1)
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+void RewriteMeeting(string oldName, string newName, DateTime newStart, int newDuration, bool onlyOnce = false)
+{
+    for (int i = 0; i < meetings.Length; i++)
+    {
+        (string name, DateTime start, int duration, string room) = meetings[i];
+  
+        if (name.Equals(oldName))
+        {
+            meetings[i].Item1 = newName;
+            meetings[i].Item2 = newStart;
+            meetings[i].Item3 = newDuration;
+
+            if (onlyOnce)
+            {
+                break;
+            }  
+        }
+    }
+
+    DumpToFile();
+}
+
+bool DoesUpdatingIntersect((string, DateTime, int, string)[] updatingMeetings, string currentName, DateTime start, int duration)
+{
+    foreach ((string name, DateTime, int, string room) updatingMeeting in updatingMeetings)
+    {
+        (string, DateTime, int, string) meeting = (updatingMeeting.name, start, duration, updatingMeeting.room);
+        
+        if (DoesIntersectWithOther(meeting, currentName, start))
+        {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+(string, DateTime, int, string)[] MeetingsByDateRange(DateTime start, DateTime end)
+{
+    (string, DateTime, int, string)[] arr = new (string, DateTime, int, string)[0];
+    
+    foreach((string name, DateTime start, int duration, string room) meeting in meetings)
+    {
+        DateTime meetingEnd = meeting.start.AddMinutes(meeting.duration);
+ 
+        if (meeting.start >= start && meetingEnd <= end)
+        {
+            Array.Resize(ref arr, arr.Length + 1);
+            arr[^1] = meeting;
+        }
+    }
+    
+    return arr;
+}
+
+static void PrintMeetings((string, DateTime, int, string)[] meetings)
+{
+    Console.WriteLine();
+    Console.WriteLine($"{"Name",-25}{"Start",-25}{"End",-25}{"Room",-25}");
+
+    foreach ((string name, DateTime start, int duration, string room) in meetings)
+    {
+        DateTime end = start.AddMinutes(duration);
+        Console.WriteLine($"{name,-25}{start,-25}{end,-25}{room,-25}");
+    }
+
+    Console.WriteLine();
+}
+
+void CoolMethod()
+{
+    Console.Clear();
+    try
+    {
+        NotImplementedMethod();

```

Comment:
you always throw `NotImplementedMethod` and catch it below, why do you need it? :)